src/components/StarWars.js

```diff
@@ -0,0 +1,182 @@
+import React, { Component } from 'react';
+import { View, Text, FlatList, ActivityIndicator } from 'react-native';
+
+import Relay, {
+  graphql,
+  QueryRenderer,
+  createRefetchContainer
+} from 'react-relay';
+
+import environment from '../environment';
+
+
+class StarWars extends Component {
+  state = {
+    isFetching: false,
+    isFetchingEnd: false,
+    page: 1,
+    dataSource: []
+  };
+  renderItem = ({ item }) => {
+    const { name, gender } = item;
+    return (
+      <View>
+        <Text>{name}</Text>
+        <Text>{gender}</Text>
+      </View>
+    );
+  }
+
+  onRefresh = () => {
+    const { isFetching } = this.state;
+
+    if (isFetching) return;
+
+    this.setState({ isFetching: true });
+
+    const refetchVariables = fragmentVariables => ({
+      ...fragmentVariables,
+    });
+
+    this.props.relay.refetch(
+      refetchVariables,
+      null,
+      () => {
+        this.setState({
+          isFetching: false,
+          isFetchingEnd: false,
+        });
+      },
+      {
+        force: true,
+      },
+    );
+  };
+
+  onEndReached = () => {
+    const { isFetchingEnd, page } = this.state;
+
+    if (isFetchingEnd === true) return;
+
+    if (page > 9) return;
+
+    this.setState({
+      isFetchingEnd: true,
+      page: page + 1,
+    });
+
+    
+    const refetchVariables = fragmentVariables => ({
+      ...fragmentVariables,
+      page: page + 1
+    });
+
+    this.props.relay.refetch(
+      refetchVariables,
+      null,
```

Comment:
missing `renderVariables` here
see this: https://facebook.github.io/relay/docs/en/refetch-container.html#arguments
its buggy, so search for `renderVariables`

Reply:
then add here:
```jsx
renderVariables,
```